src/enacl.erl

```diff
@@ -380,6 +382,29 @@ sign_open(SM, PK) ->
           end
     end.
 
+%% @doc sign_detached/2 computes a digital signature given a message and a secret key.
+%%
+%% Given a message `M' and a secret key `SK' the function will compute the digital signature `DS'.
+%% @end
+-spec sign_detached(M, SK) -> DS
+  when
+    M  :: iodata(),
+    SK :: binary(),
+    DS :: binary().
+sign_detached(M, SK) -> enacl_nif:crypto_sign_detached(M, SK).
+
+%% @doc sign_verify_detached/3 verifies the given signature against the given
+%% message for the given public key.
+%%
+%% Given a signature `SIG', a message `M', and a public key `PK', the function computes
+%% true iff the `SIG' is valid for `M' and `PK'.
+-spec sign_verify_detached(SIG, M, PK) -> boolean()
```

Comment:
An alternative is to write the spec as

```
-spec sign_verify_detached(SIG, M, PK) -> {ok, M} | {error, failed_verification}
```
because this matches `sign_open` and the other decoder paths. It also allows you to run the code without boolean blindness:

```
case sign_verify_detached(SIG, Tainted, PK) of
    {ok, Verified} -> do_something_with(Verified);
    {error, failed_verification} -> exit(argh)
end
```

The implementation is deceptively simple:

```
sign_verify_detached(SIG, M, PK) ->
    case enacl_nif:crypto_sign_verify_detached(SIG, M, PK) of
        true -> {ok, M};
        false -> {error, failed_verification}
    end.
```